addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/JobUtils.java

```diff
@@ -224,7 +227,7 @@ public static void applyParamsToObject(
         }
 
         try {
-            Method[] methods = object.getClass().getMethods();
+            Method[] methods = object.getClass().getMethods(); //Note: here the getMethods is not returning our new header setters/getters
```

Comment:
I'm sure this was handy for you, but it doesn't seem generally relevant to all uses/users.